light_curves/light_curve_classifier.md

```diff
@@ -0,0 +1,1071 @@
+#set them as indices and they won't get pivoted into
+df_lc = df_lc.set_index(["objectid", "label", "time"])
+
+#attach bandname to all the fluxes and uncertainties
+df_lc = df_lc.pivot(columns = "band")
+
+#rename the columns
+df_lc.columns = ["_".join(col) for col in df_lc.columns.values]
+
+#many of these flux columns still have a space in them from the bandnames, 
+#convert that space to underscore
+df_lc.columns = df_lc.columns.str.replace(' ', '_') 
+
+#and get rid of that index to cleanup
+df_lc = df_lc.reset_index()  
+```
+
+```{code-cell} ipython3
+#look at a single object to see what this array looks like
+ob_of_interest = 4
+singleob = df_lc[df_lc['objectid'] == ob_of_interest]
+singleob
+```
+
+### 2.7 Normalize 
+- this is normalizing across all bands
+- think this is the right place to do this, rather than interpolate over time 
+    so that the final light curves are normalized since that is the chunk of information 
+    which goes into the ML algorithms.
+- chose max and not median or mean because there are some objects where the median flux = 0.0
+    - if we did this before the interpolating, the median might be a non-zero value
+- normalizing is required so that the CLAGN and it's comparison SDSS sample don't have different flux levels.
+
+
+Idea here is that we normalize across each object.  So the algorithms will know, for example, that within one object W1 will be brighter than ZTF bands but from one object to the next, it will not know that one is brighter than the other.
+
+```{code-cell} ipython3
+# make a new column with max_r_flux for each objectid
+df_lc['max_W1'] = df_lc.groupby('objectid', sort=False)["flux_w1"].transform('max')
+
+#figure out which columns in the dataframe are flux columns
+flux_cols = [col for col in df_lc.columns if 'flux' in col]
+
+# make new normalized flux columns for all fluxes
+df_lc[flux_cols] = df_lc[flux_cols].div(df_lc['max_W1'], axis=0)
+
+#now drop max_W1 as a column so it doesn't get included as a variable in multivariate analysis
+df_lc.drop(columns = ['max_W1'], inplace = True)
+```
+
+### 2.8 Make datetime column
+https://docs.python.org/3/library/datetime.html#module-datetime
+
+SKTime wants a datetime column
+
+```{code-cell} ipython3
+#need to convert df_lc time into datetime
+mjd = df_lc.time
+
+#convert to JD
+jd = mjd_to_jd(mjd)
+
+#convert to individual components
+t = Time(jd, format = 'jd' )
+
+#t.datetime is now an array of type datetime
+#make it a column in the dataframe
+df_lc['datetime'] = t.datetime
+```
+
+### 2.9 Save this dataframe
+
+```{code-cell} ipython3
+#save this dataframe to use for the ML below so we don't have to make it every time
+parquet_savename = 'output/df_lc_ML.parquet'
+df_lc.to_parquet(parquet_savename)
+#print("file saved!")
+```
+
+### 2.10 Make into multi-index
+(which is what SKTime expects as input)
+
+```{code-cell} ipython3
+df_lc = df_lc.set_index(["objectid", "label", "datetime"])
```

Comment:
yes, I see your point that it is not needed until 4.1.  At the same time, I worked hard to get the train test split to work on the data frame with indices, so it would take me a long time to re-write that without the multiindex.  Maybe this is somewhere that I could learn from you.  I'll make it an issue and we can consider it going forward.